newspaper/article.py

```diff
@@ -160,14 +178,14 @@ def download(self, input_html=None, title=None, recursion_counter=0):
         infinite
         """
         if input_html is None:
-            try:
-                html = network.get_html_2XX_only(self.url, self.config)
-            except requests.exceptions.RequestException as e:
-                self.download_state = ArticleDownloadState.FAILED_RESPONSE
-                self.download_exception_msg = str(e)
+            parsed_url = urlparse(self.url)
+            if parsed_url.scheme == "file":
+                html = self._parse_scheme_file(parsed_url.path)
+            else:
+                html = self._parse_scheme_http()
+            if html is None:
                 log.debug('Download failed on URL %s because of %s' %
                           (self.url, self.download_exception_msg))
```

Comment:
I will approve after you add back this `return` statement, as if html is none we can skip logic the further post-processing, the return gives a bit of clarity to control flow
Nice job!!

Reply:
Updated